File: internal/bisect/bisect.go
package bisect

// Split divides a list of PR numbers into two halves for binary bisection.
// Left half gets the first ceil(n/2) elements, right half gets the rest.
// Returns (left, right). If len(prs) <= 1, right is nil.
func Split(prs []int) (left, right []int) {
	if len(prs) == 0 {
		return nil, nil
	}
	if len(prs) == 1 {
		return prs, nil
	}
	mid := (len(prs) + 1) / 2
	return prs[:mid], prs[mid:]
}

File: src/bisect.ts
/**
 * Split divides a list of PR numbers into two halves for binary bisection.
 * Left half gets the first ceil(n/2) elements, right half gets the rest.
 */
export function split(prs: number[]): [number[], number[]] {
  if (prs.length <= 1) return [[...prs], []];
  const mid = Math.ceil(prs.length / 2);
  return [prs.slice(0, mid), prs.slice(mid)];
}
